Makefile

```
@@ -83,6 +96,28 @@ all ::
 	@echo "HTML built: `ls -lAFgh $(BUILD_DIR)/manual/build/html/index.html`"
 	@echo "PDF built: `ls -lAFgh $(BUILD_DIR)/manual/build/latex/nexus.pdf`"
 
+$(BASE_CLASS_DIR)/%.nxdl.xml : $(BASE_CLASS_DIR)/$(NYAML_SUBDIR)/%.yaml
+	$(PYTHON) -m dev_tools.nyaml2nxdl.nyaml2nxdl --input-file $<
+	mv $(BASE_CLASS_DIR)/$(NYAML_SUBDIR)/$*.nxdl.xml $@
+
+$(CONTRIB_DIR)/%.nxdl.xml : $(CONTRIB_DIR)/$(NYAML_SUBDIR)/%.yaml
+	$(PYTHON) -m dev_tools.nyaml2nxdl.nyaml2nxdl --input-file $<
+	mv $(CONTRIB_DIR)/$(NYAML_SUBDIR)/$*.nxdl.xml $@
+
+$(APPDEF_DIR)/%.nxdl.xml : $(APPDEF_DIR)/$(NYAML_SUBDIR)/%.yaml
+	$(PYTHON) -m dev_tools.nyaml2nxdl.nyaml2nxdl --input-file $<
+	mv $(APPDEF_DIR)/$(NYAML_SUBDIR)/$*.nxdl.xml $@
+
+NXDLS := $(foreach dir,$(NXDL_DIRS),$(wildcard $(dir)/*.nxdl.xml))
```

Just use sum of other variables

---

I don't know exactly what you would like to have, something like this?

```suggestion
NXDLS := $(foreach dir, $(BASE_CLASS_DIR) $(CONTRIB_DIR) $(APPDEF_DIR), $(wildcard $(dir)/*.nxdl.xml))
```

---

It's simply `$(NXDL_BC) + $(NXDL_APPDEF) + $(NXDL_CONTRIB)`

---

Ah yes, that makes sense (wasn't aware of this `+` operator though). I changed it accordingly. Thanks!